airbyte-webapp/src/components/connectorBuilder/types.ts

```diff
@@ -607,3 +674,246 @@ export const convertToManifest = (values: BuilderFormValues): ConnectorManifest
+    );
+  }
+
+  const inlineSchemaLoader = manifestSchemaLoader as InlineSchemaLoader;
+
+  if (inlineSchemaLoader.schema) {
+    return JSON.stringify(inlineSchemaLoader.schema);
+  }
+  return undefined;
+}
+
+const interpolatedConfigValueRegex = /^{{config\[('|"+)(.+)('|"+)\]}}$/;
+
+function isInterpolatedConfigKey(str: string): boolean {
+  const noWhitespaceString = str.replace(/\s/g, "");
+  return interpolatedConfigValueRegex.test(noWhitespaceString);
+}
+
+function extractInterpolatedConfigKey(str: string): string | undefined {
+  const noWhitespaceString = str.replace(/\s/g, "");
+  const regexResult = interpolatedConfigValueRegex.exec(noWhitespaceString);
+  console.log("regexResult", regexResult);
+  if (regexResult === null) {
+    return undefined;
+  }
+  return regexResult[2];
+}
+
+function manifestAuthenticatorToBuilder(
+  manifestAuthenticator: HttpRequesterAuthenticator | undefined,
+  streamName?: string
+): BuilderFormAuthenticator {
+  let builderAuthenticator: BuilderFormAuthenticator;
+  if (manifestAuthenticator === undefined) {
+    builderAuthenticator = {
+      type: "NoAuth",
+    };
+  } else if (manifestAuthenticator.type === "CustomAuthenticator") {
+    throw new ManifestCompatibilityError(streamName, "uses a CustomAuthenticator");
+  } else if (manifestAuthenticator.type === "OAuthAuthenticator") {
+    builderAuthenticator = {
+      ...manifestAuthenticator,
+      refresh_request_body: Object.entries(manifestAuthenticator.refresh_request_body ?? {}),
+    };
+  } else {
+    builderAuthenticator = manifestAuthenticator;
+  }
+
+  // verify that all auth keys which require a user input have a {{config[]}} value
+
+  const userInputAuthKeys = Object.keys(authTypeToKeyToInferredInput[builderAuthenticator.type]) as Array<
+    keyof typeof builderAuthenticator
+  >;
+
+  for (const userInputAuthKey of userInputAuthKeys) {
+    if (!isInterpolatedConfigKey(builderAuthenticator[userInputAuthKey])) {
+      throw new ManifestCompatibilityError(
+        undefined,
+        `Authenticator's ${userInputAuthKey} value must be of the form {{ config['key'] }}`
+      );
+    }
+  }
+
+  return builderAuthenticator;
+}
+
+function manifestSpecAndAuthToBuilder(
+  manifestSpec: Spec | undefined,
+  manifestAuthenticator: HttpRequesterAuthenticator | undefined
+): { inputs: BuilderFormValues["inputs"]; auth: BuilderFormAuthenticator } {
+  const result: { inputs: BuilderFormValues["inputs"]; auth: BuilderFormAuthenticator } = {
+    inputs: [],
+    auth: manifestAuthenticatorToBuilder(manifestAuthenticator),
+  };
+
+  if (manifestSpec === undefined) {
+    return result;
+  }
+
+  const required = manifestSpec.connection_specification.required as string[];
+  result.inputs = Object.entries(
+    manifestSpec.connection_specification.properties as Record<string, AirbyteJSONSchema>
+  ).flatMap(([key, definition]) => {
+    if (Object.keys(authTypeToKeyToInferredInput[result.auth.type]).includes(key)) {
```

Comment:
looks like the keys of that object are the properties of the authenticator object, not the name of the input (in this case it's causing a mismatch between api_key and api_token). Should probably be
```
if (Object.values(authTypeToKeyToInferredInput[result.auth.type])
        .map((input) => input.key).includes(key)) {
```

Otherwise you will end up with two api_keys when switching around